### PR TITLE
ripcord: add darwin app bundle

### PR DESCRIPTION
**Pull request by mikroskeem**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix [hunk 1]
```diff
@@ -8,29 +8,44 @@ mkDerivation rec {
...
+  in if (stdenv.isDarwin) then appbundle else appimageTools.extract {
     name = "${pname}-${version}";
     src = appimage;
   };
 
-  nativeBuildInputs = [ autoPatchelfHook desktop-file-utils imagemagick ];
-  buildInputs = [ libsodium libopus libGL alsa-lib ]
-    ++ [ qtbase qtsvg qtmultimedia qtwebsockets qtimageformats ]
-    ++ (with xorg; [ libX11 libXScrnSaver libXcursor xkeyboardconfig ]);
+  nativeBuildInputs = lib.optionals stdenv.isLinux [
```
Comment (veprbl) [inline pkgs/applications/networking/instant-messengers/ripcord/default.nix:27]: With so many differences between darwin and linux, might as well define it in a separate mkDerivation call

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix [hunk 2]
```diff
@@ -53,6 +68,9 @@ mkDerivation rec {
       --prefix LD_LIBRARY_PATH ":" "${xorg.libXcursor}/lib" \
       --prefix QT_XKB_CONFIG_ROOT ":" "${xorg.xkeyboardconfig}/share/X11/xkb" \
       --set RIPCORD_ALLOW_UPDATES 0
+  '' + lib.optionalString stdenv.isDarwin ''
+    mkdir -p $out/Applications
+    cp -r $src/Ripcord.app $out/Applications/
```
Comment (veprbl) [inline pkgs/applications/networking/instant-messengers/ripcord/default.nix:73]:
```suggestion
    cp -r $src/Ripcord.app $out/Applications/
    '' + ''
```

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix (hunk 1 above)
Comment (mikroskeem) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix:27]: Sure

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix (hunk 2 above)
Comment (mikroskeem) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix:73]: Good catch, thanks

**Review by veprbl — APPROVED** [review]

**Comment by mikroskeem** [comment]
Last batch of suggestions applied as well

**Pull request merged**